docs/tutorialkit.dev/astro.config.ts

```diff
@@ -67,6 +67,10 @@ export default defineConfig({
               label: 'Overriding Components',
               link: '/guides/overriding-components/',
             },
+            {
+              label: 'Using low level APIs',
+              link: '/guides/using-low-level-apis/',
+            },
```

Comment:
This page will describe how to use `tutorialkit:store` and `tutorialkit:core`. What would be a good name for this page? Maybe something like:

- TutorialKit Node APIs
- Store and Code APIs

Feel free to suggest better name. 🙌

Reply:
So I would actually have more than one page:
* One under `reference` which could be called `TutorialKit API` which describes what is exposed by each module and maybe showing the functions similarly to https://webcontainers.io/api.
* One (but ideally more) under guides, that are _literal_ ["How to"](https://diataxis.fr/how-to-guides/)  guides explaining how to solve a specific problems that people have been running into.

So examples of what I think we should have (in no particular order):
- How to write custom components that access the tutorial state? _(basically what you've written so far)_
- How to programmatically write to one of the terminal? _(example component doing it via a button)_
- How to check the lesson was solved and congratz the user? _(your task example + https://github.com/stackblitz/tutorialkit/discussions/205)_
- How to watch a file? _(using `tutorialStore.onDocumentChanged`)_

Reply:
Awesome, thanks! This makes the documentation task much clearer 😄 

So we will have these pages:

- Reference
  - TutorialKit API
    - Tutorial Store: describes all public APIs. We might want to mark all other ones with `@internal` JSDoc and use `stripInternal` in `tsconfig.json`.
    - Tutorial Core: describes how to access `webcontainer` and has link to webcontainers.io

- Guide
  - How to pages:
    - `Access tutorial state`. Uses `tutorialkit:store`. Do we show how to access file contents here like in the `HelpButton.tsx` or `TaskList.tsx`? 
    - `Write to terminal`. How is this done? `webcontainer.spawn`?
    - `Congrats the user when lesson is solved`. How to do this? There's no `lessonSolved`.
    - `How to watch a file`. `tutorialStore.onDocumentChanged`

Reply:
@Nemikolh I've just started to write the reference part. So something like this? 

<details>

```mdx
---
title: TutorialKit API
description: Use TutorialKit's lower level APIs
---

TutorialKit exposes low level APIs that authors can utilize to provide highly custom experience in their tutorials.

## Tutorial Store

You can access Tutorial Store by importing the `tutorialkit:store` entrypoint.

```ts
import tutorialStore from "tutorialkit:store";
```

### Common types

- `ReadableAtom` from [`nanostores`](https://www.npmjs.com/package/nanostores)

### Properties

#### `previews`

Instances of the preview tabs.

Type: `ReadableAtom<PreviewInfo[]>`

```ts
import type { PreviewSchema } from '@tutorialkit/types';

class PreviewInfo {
    readonly portInfo: PortInfo;
    title?: string;
    pathname?: string;
    get url(): string | undefined;
    get port(): number;
    get baseUrl(): string | undefined;
    get ready(): boolean;
    static parse(preview: Exclude<PreviewSchema, boolean>[0]): Preview;
    static equals(a: PreviewInfo, b: PreviewInfo): boolean;
}

interface Preview {
    port: number;
    pathname?: string;
    title?: string;
}
```

#### `terminalConfig`

Type: `ReadableAtom<TerminalConfig>`

#### `editorConfig`

Type: `ReadableAtom<EditorConfig>`

#### `currentDocument`

Type: `ReadableAtom<EditorDocument | undefined>`

#### `bootStatus`

Type: `ReadableAtom<BootStatus>`

#### `documents`

Type: `ReadableAtom<EditorDocuments>`

#### `files`

Type: `ReadableAtom<FileDescriptor[]>`

#### `template`

Type: `Files | undefined`

#### `selectedFile`

Type: `ReadableAtom<string | undefined>`

#### `lesson`

Type: `Readonly<Lesson> | undefined`

#### `themeRef`

Type: `ReadableAtom<unknown>`


### Methods


## Tutorial Core

```ts
import * as core from "tutorialkit:core";
```

...
```

</details>

Making all of this stuff part of public API is maybe too much. We'll need to always avoid making breaking changes in these APIs.